src/obsidian_ai_hub/coding/backend.py

```
@@ -79,6 +79,94 @@ def check_dirty_tree(repo_path: str | Path) -> tuple[bool, str]:
+        branch = proc.stdout.strip()
+        if not branch:
+            # Fallback to commit SHA / HEAD description if detached
+            rev_proc = subprocess.run(
+                ["git", "rev-parse", "--short", "HEAD"],
+                cwd=path,
+                capture_output=True,
+                text=True,
+            )
+            if rev_proc.returncode == 0:
+                branch = rev_proc.stdout.strip()
+    except Exception as exc:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)